### PR TITLE
expression: Optimize code if else branch logic

### PR DESCRIPTION
**Pull request by Zh1Cheung**
### What problem does this PR solve?

Issue Number:   #29518 

Make the code more Readable

### What is changed and how it works?

Reduce if else branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by TszKitLo40** [comment]
You also need to sign the `Contributor License Agreement` to pass the check.

File: server/server.go [hunk 1]
```diff
@@ -304,25 +302,28 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
...
 	sockStat, err := os.Stat(socket)
-	if err == nil {
-		if sockStat.Mode().Type() != os.ModeSocket {
-			return fmt.Errorf(
-				"the specified socket file %s is a %s instead of a socket file",
-				socket, sockStat.Mode().String())
-		}
+	// Todo:According to the previous logic, When it's occur error will return nil?
+	if err != nil {
+		//return fmt.Errorf("stat socket file %s failed: %v", socket, err)
```
Comment (xhebox) [inline server/server.go:307]: Remove this line, plz.

Yes it is nil, since this is an deleting operation, and it is silent about the errors.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Review by TszKitLo40 — APPROVED** [review]
LGTM

**Comment by Zh1Cheung** [comment]
> Rest LGTM

done

**Review by xhebox — APPROVED** [review]

**Pull request merged**